mltsp/build_model.py

```diff
@@ -31,21 +32,71 @@ def rectangularize_featureset(featureset):
     return feature_df.loc[featureset.name]  # preserve original row ordering
 
 
+def fit_model_optimize_hyperparams(data, targets, model, model_params,
+                                   params_to_optimize):
```

Comment:
From conversation w/ @stefanv: `model_params` and `params_to_optimize` both `dict`s instead of `dict` and list of names.